offer-manager-domain/src/test/groovy/life/qbic/business/ProductFactory.groovy

```diff
@@ -86,6 +87,15 @@ class ProductFactory {
                         runningNumber,
                         serviceProvider) as T
                 break
+            case ExternalServiceProduct:
+                return new ExternalServiceProduct(name,
+                        description,
+                        internalPrice,
+                        externalPrice,
+                        productUnit,
+                        runningNumber,
+                        serviceProvider) as T
+                break
```

Comment:
NIT:
can this be refactored for the whole code to create a product (in an external method) and for each case call the method and cast it to the respective type?

Reply:
I am not sure how this will work, do you have an example?

Reply:
Nvm I forgot that a product is abstract